Update citation (#606)
adapter-hub/adapters: docs/index.rst

Citation
========

> If you use this library practically in your work, please consider citing our library paper: `Adapters: A Unified Library for Parameter-Efficient and Modular Transfer Learning <https://arxiv.org/abs/2311.11077>`_

Reviewer: ```suggestion
If you use the _Adapters_ library in your work, please consider citing our library paper `Adapters: A Unified Library for Parameter-Efficient and Modular Transfer Learning <https://arxiv.org/abs/2311.11077)>`
```

.. code-block:: bibtex

   @misc{poth2023adapters,
         title={Adapters: A Unified Library for Parameter-Efficient and Modular Transfer Learning}, 
         author={Clifton Poth and Hannah Sterz and Indraneil Paul and Sukannya Purkayastha and Leon Engländer and Timo Imhof and Ivan Vulić and Sebastian Ruder and Iryna Gurevych and Jonas Pfeiffer},
         year={2023},
         eprint={2311.11077},
         archivePrefix={arXiv},
         primaryClass={cs.CL}
   }

> Alternatively, if you rather want to refer to the predecessor `adapter-transformers` or reference models from the Hub and the documentation, please consider citing our initial research paper: `AdapterHub: A Framework for Adapting Transformers <https://arxiv.org/abs/2007.07779>`_

Reviewer: ```suggestion
Alternatively, for the predecessor `adapter-transformers`, the Hub infrastructure and adapters uploaded by the AdapterHub team, please consider citing our initial paper: `AdapterHub: A Framework for Adapting Transformers <https://arxiv.org/abs/2007.07779>`_
```
